packages/pds/src/sequencer/events.ts

```diff
@@ -1,74 +1,61 @@
-import { CID } from 'multiformats/cid'
 import { z } from 'zod'
-import { cborEncode, schema } from '@atproto/common'
-import {
-  BlockMap,
-  CidSet,
-  CommitData,
-  WriteOpAction,
-  blocksToCarFile,
-} from '@atproto/repo'
+import { cborEncode, noUndefinedVals, schema } from '@atproto/common'
+import { BlockMap, blocksToCarFile } from '@atproto/repo'
 import { AccountStatus } from '../account-manager'
-import { PreparedWrite } from '../repo'
+import { CommitDataWithOps } from '../repo'
 import { RepoSeqInsert } from './db'
 
 export const formatSeqCommit = async (
   did: string,
-  commitData: CommitData,
-  writes: PreparedWrite[],
+  commitData: CommitDataWithOps,
 ): Promise<RepoSeqInsert> => {
-  let tooBig: boolean
-  const ops: CommitEvtOp[] = []
-  const blobs = new CidSet()
-  let carSlice: Uint8Array
-
   const blocksToSend = new BlockMap()
   blocksToSend.addMap(commitData.newBlocks)
   blocksToSend.addMap(commitData.relevantBlocks)
 
-  // max 200 ops or 1MB of data
-  if (writes.length > 200 || blocksToSend.byteSize > 1000000) {
-    tooBig = true
+  let evt: CommitEvt
+
+  // If event is too big (max 200 ops or 1MB of data)
+  if (commitData.ops.length > 200 || blocksToSend.byteSize > 1000000) {
     const justRoot = new BlockMap()
     const rootBlock = blocksToSend.get(commitData.cid)
     if (rootBlock) {
       justRoot.set(commitData.cid, rootBlock)
     }
-    carSlice = await blocksToCarFile(commitData.cid, justRoot)
+
+    evt = {
+      rebase: false,
+      tooBig: true,
+      repo: did,
+      commit: commitData.cid,
+      prev: commitData.prev,
+      rev: commitData.rev,
+      since: commitData.since,
+      blocks: await blocksToCarFile(commitData.cid, justRoot),
+      ops: [],
+      blobs: [],
+      prevData: commitData.prevData ?? undefined,
+    }
   } else {
-    tooBig = false
-    for (const w of writes) {
-      const path = w.uri.collection + '/' + w.uri.rkey
-      let cid: CID | null
-      if (w.action === WriteOpAction.Delete) {
-        cid = null
-      } else {
-        cid = w.cid
-        w.blobs.forEach((blob) => {
-          blobs.add(blob.cid)
-        })
-      }
-      ops.push({ action: w.action, path, cid })
+    evt = {
+      rebase: false,
+      tooBig: false,
+      repo: did,
+      commit: commitData.cid,
+      prev: commitData.prev,
+      rev: commitData.rev,
+      since: commitData.since,
+      blocks: await blocksToCarFile(commitData.cid, blocksToSend),
+      ops: commitData.ops,
+      blobs: commitData.blobs.toList(),
```

Comment:
I believe this can be left empty as part of deprecation.

Reply:
I'll do that in the follow up PR where we fully switch over to sync1.1 semantics